src/main/java/seedu/address/logic/parser/DeleteCommandParser.java

```diff
@@ -16,13 +14,13 @@ public class DeleteCommandParser implements Parser<DeleteCommand> {
      * and returns a DeleteCommand object for execution.
      * @throws ParseException if the user input does not conform the expected format
      */
-    public DeleteCommand parse(String args) throws ParseException {
+    public DeleteCommand parse(String args) {
         try {
             Index index = ParserUtil.parseIndex(args);
             return new DeleteCommand(index);
         } catch (ParseException pe) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteCommand.MESSAGE_USAGE), pe);
+            Index placeholderIndex = null;
+            return new DeleteCommand(placeholderIndex, pe);
```

Comment:
For the catch clause, I think it is not right to return from the function. We still have to throw an exception regardless. Here is a good read from [stackoverflow](https://stackoverflow.com/questions/36676929/can-i-return-false-from-a-try-and-catch-instead-of-throwing-exception)